examples/metricq-sink-dummy/src/dummy_sink.cpp

```diff
@@ -98,6 +102,21 @@ void DummySink::on_data_channel_ready()
             return metricq::Timer::TimerResult::repeat;
         },
         std::chrono::seconds(10));
+
+    if (timeout_)
+    {
+        timeout_timer_.start(
+            [this](std::error_code error) {
+                if (!error)
+                {
+                    Log::error() << "Data timeout! Didn't receive data in " << timeout_ << " s.";
```

Comment:
```suggestion
                    Log::error() << "Data timeout! Didn't receive data in " << this->timeout_ << " s.";
```

There are compilers in the wild, which are confused about member variables without the `this`, when `this` was captured in a lambda.